app/proxy/SWProxy.js

```diff
@@ -17,114 +21,27 @@ class SWProxy extends EventEmitter {
-            self.emit('apiCommand', reqData, respData);
-            delete parsedRequests[command];
-          }
-        });
-      }
+    proxy.onError(function(ctx, err, errorKind) {
+      // ctx may be null
+      var url = ctx && ctx.clientToProxyRequest ? ctx.clientToProxyRequest.url : '';
+      console.error(errorKind + ' on ' + url + ':', err);
     });
 
-    this.proxy.on('error', (error, req, resp) => {
-      resp.writeHead(500, {
-        'Content-Type': 'text/plain'
-      });
+    proxy.onRequest(function(ctx, callback) {
+      if (ctx.clientToProxyRequest.headers.host == 'www.google.com' && ctx.clientToProxyRequest.url.indexOf('/search') == 0) {
+        ctx.use(Proxy.gunzip);
 
-      resp.end('Something went wrong.');
-    });
-
-    this.httpServer = http
-      .createServer((req, resp) => {
-        // Request has been intercepted from game client
-        let reqChunks = [];
-        if (req.url.indexOf('qpyou.cn/api/gateway_c2.php') >= 0) {
-          req.on('data', chunk => {
-            reqChunks.push(chunk);
-          });
-          req.on('end', () => {
-            // Parse the request
-            let reqData;
-            try {
-              reqData = decrypt_request(reqChunks.join());
-            } catch (e) {
-              // Error decrypting the data, log and do not fire an event
-              self.log({ type: 'debug', source: 'proxy', message: `Error decrypting request data - ignoring. ${e}` });
-              return;
-            }
-
-            const { command } = reqData;
-
-            // Add command request to an object so we can handle multiple requests at a time
-            parsedRequests[command] = reqData;
-          });
-        }
-
-        this.proxy.web(req, resp, { target: req.url, prependPath: false });
-      })
-      .listen(port, () => {
-        this.log({ type: 'info', source: 'proxy', message: `Now listening on port ${port}` });
-        if (process.env.autostart) {
-          console.log(`SW Exporter Proxy is listening on port ${port}`);
-        }
-        win.webContents.send('proxyStarted');
-      });
-
-    this.httpServer.on('error', e => {
-      if (e.code === 'EADDRINUSE') {
-        self.log({ type: 'warning', source: 'proxy', message: 'Port is in use from another process. Try another port.' });
+        ctx.onResponseData(function(ctx, chunk, callback) {
+          chunk = new Buffer(chunk.toString().replace(/<a.*?<\/a>/g, '<a>Pwned!</a>'));
+          return callback(null, chunk);
+        });
       }
+      return callback();
     });
 
-    this.httpServer.on('connect', (req, socket) => {
-      const serverUrl = url.parse(`https://${req.url}`);
-
-      const srvSocket = net.connect(serverUrl.port, serverUrl.hostname, () => {
-        socket.write('HTTP/1.1 200 Connection Established\r\n' + 'Proxy-agent: Node-Proxy\r\n' + '\r\n');
-        srvSocket.pipe(socket);
-        socket.pipe(srvSocket);
-      });
+    proxy.listen({ port})
```

Comment:
did you try `{port: port}`?  I'm not a NodeJS guy so I'm less confident about optional argument syntax.

Reply:
It also accepts a `host: <host>` that you 
might want to hardcode to `0.0.0.0` while testing.

Reply:
`{ port }`just means `{ port: port }` in JS.
Although I doubt that the hostname is necessary, I tried it. Didn't change anything.